spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -705,10 +705,14 @@ def set_color_scheme(self, color_scheme, reset=True):
         )
 
     def update_syspath(self, path_dict, new_path_dict):
-        """Update sys.path contents on kernel."""
-        self.call_kernel(
-            interrupt=True,
-            blocking=False).update_syspath(path_dict, new_path_dict)
+        """Update sys.path contents in the kernel."""
+        # Prevent error when the kernel is not available and users open/close
+        # projects or use the Python path manager.
+        # Fixes spyder-ide/spyder#21563
+        if self.kernel_handler is not None:
+            self.call_kernel(interrupt=True, blocking=False).update_syspath(
+                path_dict, new_path_dict
+            )
```

Comment:
Shouldn't the `sys.path` contents that was updated be saved somewhere to properly apply it when `kernel_handler` is available? Or maybe there is no possiblitiy that at some point later on the `kernel_handler` becomes available?

Reply:
I thought about that and concluded it's not necessary because this error can only happen when the kernel is not available (because it errored at startup or the Spyder-kernels version is wrong).